### PR TITLE
[Vectorized](function) support avg_weighted/percentile_array/topn_wei…

### PR DESCRIPTION
**Pull request by zhangstar333**
…ghted functions

# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.28 seconds
 load time: 511 seconds
 storage size: 17177715881 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221111140145_clickbench_pr_44523.html

File: be/src/vec/aggregate_functions/aggregate_function_topn.h [hunk 1]
```diff
@@ -228,6 +230,34 @@ struct AggregateFunctionTopNImplArray {
...
+        if constexpr (has_default_param) {
+            place.set_paramenters(
+                    static_cast<const ColumnInt32*>(columns[2])->get_element(row_num),
+                    static_cast<const ColumnInt32*>(columns[3])->get_element(row_num));
+
+        } else {
+            place.set_paramenters(
+                    static_cast<const ColumnInt32*>(columns[2])->get_element(row_num));
+        }
+        if constexpr (std::is_same_v<T, std::string>) {
```
Comment (HappenLee) [inline be/src/vec/aggregate_functions/aggregate_function_topn.h:249]: here should do not use string. use `StringRef` or `StringValue` replace

File: be/src/vec/aggregate_functions/aggregate_function_topn.h [hunk 2]
```diff
@@ -228,6 +230,34 @@ struct AggregateFunctionTopNImplArray {
...
+            StringRef ref = static_cast<const ColumnString&>(*columns[0]).get_data_at(row_num);
+            auto weight = static_cast<const ColumnVector<Int64>&>(*columns[1]).get_data()[row_num];
+            place.add(std::string(ref.data, ref.size), weight);
```
Comment (HappenLee) [inline be/src/vec/aggregate_functions/aggregate_function_topn.h:252]: same as up, here cause useless mem copy

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**